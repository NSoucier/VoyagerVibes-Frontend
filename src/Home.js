import React from "react";
import { Button } from "reactstrap";
import { Link, useNavigate } from "react-router-dom";
import './Home.css';

function Home() {
    const navigate = useNavigate();

    // redirect to login page
    function handleClick() {
        navigate('/login')
    }

    return (
        <>
         <h2 className="welcome">Let the voyage begin!</h2>

         {( localStorage.user ) ?
         (<div>
            <p className="message">Where will you go next, {localStorage.user}?</p>
            <p className="message"><Link to='/explore'>Explore</Link> now!</p>
        </div>
         )
         : 
         (<Button color="success" id="home-login" onClick={handleClick}>Login</Button>)}
        </>
    )
}

export default Home;