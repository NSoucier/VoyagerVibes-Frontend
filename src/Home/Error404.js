function Error404() {
    return (
        <h4 id="error404">Test, this page could not be found (Error 404).</h4>
    )
}

export default Error404;