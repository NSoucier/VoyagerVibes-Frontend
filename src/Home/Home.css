.welcome {
    color: rgba(128, 0, 128, 0.636);
    font-family: "Pacifico", cursive;
    font-weight: 400;
    font-size: 3rem;
    font-style: normal;
    margin-top: 5%;
}

.message {
    font-size: 1.5rem;
    margin-top: 1rem;
}

#home-login {
    margin-top: 1rem;
    background-color: rgba(0, 128, 0);
}

#footer {
    position: fixed;
    bottom: 0;
}