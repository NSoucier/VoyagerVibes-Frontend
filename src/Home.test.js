import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Home from "./Home";
import { Route, Routes, MemoryRouter, BrowserRouter } from "react-router-dom";

// smoke test
it("renders without crashing", async function () {
        render(
            <BrowserRouter>
                <Home />
            </BrowserRouter>
          );
});

// snapshot test
it("matches snapshot", function() {
    const {asFragment} = render(
        <BrowserRouter>
            <Home />
        </BrowserRouter>
      );
    expect(asFragment()).toMatchSnapshot();
});

it("displays a login button", function() {
    const {container} = render(
        <BrowserRouter>
            <Home />
        </BrowserRouter>
      );

    expect(container.querySelector('#home-login')).toBeInTheDocument();
});

it("displays a welcome message to user", function() {
    localStorage.user = 'testUser'
    const {container} = render(
        <BrowserRouter>
            <Home />
        </BrowserRouter>
      );

    expect(container.querySelector('.message')).toBeInTheDocument();
    localStorage.removeItem('user');
});