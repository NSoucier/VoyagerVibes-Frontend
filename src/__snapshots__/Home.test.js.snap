// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2
    class="welcome"
  >
    Let the voyage begin!
  </h2>
  <button
    class="btn btn-success"
    id="home-login"
    type="button"
  >
    Login
  </button>
</DocumentFragment>
`;
